Sources/Amplitude/Amplitude.m

```diff
@@ -1046,6 +1008,8 @@ - (void)enterForeground {
         [self refreshDynamicConfig];
         [self startOrContinueSessionNSNumber:now];
         self->_inForeground = YES;
+        self->_eventsBuffer = [AMPStorage getEventsFromDisk:[AMPStorage getDefaultEventsFile:self.instanceName]];
```

Comment:
One thing just came to mind, would there be any case, when we call `getEventsFromDisk` we already finish the file, then we logEvent and put the event in file? And if this happens, would the file break?

Reply:
Not that I know of, given the ways that these methods are called. Take a look at enterBackground, enterForeground.